flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -612,48 +614,56 @@ private static void setBigTextStyle(NotificationDetails notificationDetails, Not
-                if (notificationDetails.playSound) {
+            if ((notificationChannel == null && (notificationChannelDetails.channelAction == null || notificationChannelDetails.channelAction == NotificationChannelAction.CreateIfNotExists)) || (notificationChannel != null && notificationChannelDetails.channelAction == NotificationChannelAction.Update)) {
+                notificationChannel = new NotificationChannel(notificationChannelDetails.id, notificationChannelDetails.name, notificationChannelDetails.importance);
+                notificationChannel.setDescription(notificationChannelDetails.description);
+                if (notificationChannelDetails.playSound) {
                     AudioAttributes audioAttributes = new AudioAttributes.Builder().setUsage(AudioAttributes.USAGE_NOTIFICATION).build();
-                    Uri uri = retrieveSoundResourceUri(context, notificationDetails);
+                    Uri uri = retrieveSoundResourceUri(context, notificationChannelDetails);
                     notificationChannel.setSound(uri, audioAttributes);
                 } else {
                     notificationChannel.setSound(null, null);
                 }
-                notificationChannel.enableVibration(BooleanUtils.getValue(notificationDetails.enableVibration));
-                if (notificationDetails.vibrationPattern != null && notificationDetails.vibrationPattern.length > 0) {
-                    notificationChannel.setVibrationPattern(notificationDetails.vibrationPattern);
+                notificationChannel.enableVibration(BooleanUtils.getValue(notificationChannelDetails.enableVibration));
+                if (notificationChannelDetails.vibrationPattern != null && notificationChannelDetails.vibrationPattern.length > 0) {
+                    notificationChannel.setVibrationPattern(notificationChannelDetails.vibrationPattern);
                 }
-                boolean enableLights = BooleanUtils.getValue(notificationDetails.enableLights);
+                boolean enableLights = BooleanUtils.getValue(notificationChannelDetails.enableLights);
                 notificationChannel.enableLights(enableLights);
-                if (enableLights && notificationDetails.ledColor != null) {
-                    notificationChannel.setLightColor(notificationDetails.ledColor);
+                if (enableLights && notificationChannelDetails.ledColor != null) {
+                    notificationChannel.setLightColor(notificationChannelDetails.ledColor);
                 }
-                notificationChannel.setShowBadge(BooleanUtils.getValue(notificationDetails.channelShowBadge));
+                notificationChannel.setShowBadge(BooleanUtils.getValue(notificationChannelDetails.showBadge));
                 notificationManager.createNotificationChannel(notificationChannel);
             }
         }
     }
 
     private static Uri retrieveSoundResourceUri(Context context, NotificationDetails notificationDetails) {
```

Comment:
if these methods are redirecting to the new `retrieveSoundResourceUri(Context context, String sound, SoundSource soundSource)` method you've defined then we could get rid of the these two (i.e. `retrieveSoundResourceUri(Context context, NotificationDetails notificationDetails)` and `retrieveSoundResourceUri(Context context, NotificationChannelDetails notificationChannelDetails)`